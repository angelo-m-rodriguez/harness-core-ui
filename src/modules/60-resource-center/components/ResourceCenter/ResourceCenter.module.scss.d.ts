/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly resourceCenter: string
  readonly allregions: string
  readonly middleregion: string
  readonly myticket: string
  readonly bottombutton: string
  readonly text: string
}
export default styles
