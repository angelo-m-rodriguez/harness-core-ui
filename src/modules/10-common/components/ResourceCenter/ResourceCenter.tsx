/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { Button, ButtonVariation, Color, FontVariation, Icon, IconName, Layout, Text } from '@wings-software/uicore'
import { Drawer, Position } from '@blueprintjs/core'
import cx from 'classnames'
import { useStrings } from 'framework/strings'
import resourceImage from './images/resource-center.png'
import css from './ResourceCenter.module.scss'

const getButton = (buttonText: string, buttonIcon: string, link: string): JSX.Element => {
  return (
    <a href={link} rel="noreferrer" target="_blank">
      <Layout.Vertical
        flex={{ align: 'center-center' }}
        spacing="small"
        padding={'small'}
        className={cx(css.bottombutton)}
      >
        <Icon name={buttonIcon as IconName} size={24} color={Color.WHITE} />
        <Text font={{ variation: FontVariation.BODY2 }} padding={{ bottom: 'xsmall' }} color={Color.PRIMARY_3}>
          {buttonText}
        </Text>
      </Layout.Vertical>
    </a>
  )
}

const menuItems = (title: string, description: string): JSX.Element => {
  return (
    <>
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H4 }} padding={{ bottom: 'xsmall' }} color={Color.PRIMARY_3}>
          {title}
        </Text>
        <Text font={{ variation: FontVariation.BODY2 }} padding={{ bottom: 'xsmall' }} color={Color.WHITE}>
          {description}
        </Text>
      </Layout.Vertical>
      <Button icon="chevron-right" variation={ButtonVariation.ICON} disabled />
    </>
  )
}

export const ResourceCenter = (): React.ReactElement => {
  const { getString } = useStrings()

  const [show, setShow] = useState<boolean>(false)

  if (!show) {
    return (
      <Icon
        name={'question'}
        onClick={e => {
          e.stopPropagation()
          e.preventDefault()
          setShow(true)
        }}
        data-testid="question"
        className={css.question}
      />
    )
  }

  return (
    <Drawer
      autoFocus
      enforceFocus
      hasBackdrop
      usePortal
      canOutsideClickClose
      canEscapeKeyClose
      position={Position.BOTTOM_LEFT}
      isOpen={show}
      onClose={() => {
        setShow(false)
      }}
      className={css.resourceCenter}
    >
      <Layout.Vertical>
        <Layout.Vertical padding={'xlarge'} flex={{ alignItems: 'baseline' }}>
          <Layout.Horizontal
            padding={{ bottom: 'medium' }}
            className={css.title}
            flex={{ justifyContent: 'space-between', alignItems: 'baseline' }}
          >
            <Text color={Color.WHITE}>{getString('common.resourceCenter.title')}</Text>
            <Button
              icon={'cross'}
              variation={ButtonVariation.ICON}
              onClick={e => {
                e.stopPropagation()
                e.preventDefault()
                setShow(false)
              }}
            />
          </Layout.Horizontal>
          <img src={resourceImage} height={106} alt={'Resource center image'} />
        </Layout.Vertical>
        <Layout.Vertical padding={'xlarge'} className={css.middleregion}>
          <Layout.Horizontal
            padding={{ bottom: 'medium' }}
            flex={{ justifyContent: 'space-between' }}
            className={css.myticket}
          >
            {menuItems(
              getString('common.resourceCenter.ticketmenu.tickets'),
              getString('common.resourceCenter.ticketmenu.ticketsDesc')
            )}
          </Layout.Horizontal>
          <Layout.Horizontal padding={{ top: 'medium' }} flex={{ justifyContent: 'space-between' }}>
            {menuItems(
              getString('common.resourceCenter.ticketmenu.submit'),
              getString('common.resourceCenter.ticketmenu.submitDesc')
            )}
          </Layout.Horizontal>
        </Layout.Vertical>
        <Layout.Vertical padding={'xlarge'}>
          <Text font={{ variation: FontVariation.BODY2 }} padding={{ bottom: 'medium' }} color={Color.WHITE}>
            {getString('common.resourceCenter.bottomlayout.desc')}
          </Text>
          <Layout.Horizontal flex={{ justifyContent: 'space-around' }}>
            {getButton(getString('search'), 'thinner-search', 'https://harness.io/search/')}
            {getButton(
              getString('common.resourceCenter.bottomlayout.docs.text'),
              'resource-center-docs-icon',
              'https://docs.harness.io/'
            )}
            {getButton(
              getString('common.resourceCenter.bottomlayout.community.text'),
              'resource-center-community-icon',
              'https://community.harness.io/'
            )}
            {getButton(
              getString('common.resourceCenter.bottomlayout.sitestatus.text'),
              'right-bar-notification',
              'https://status.harness.io/'
            )}
          </Layout.Horizontal>
        </Layout.Vertical>
      </Layout.Vertical>
    </Drawer>
  )
}
