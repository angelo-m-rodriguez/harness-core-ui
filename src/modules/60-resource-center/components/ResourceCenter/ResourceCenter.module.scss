$col1: var(--primary-9-dark);

.resourceCenter {
  width: 440px;
  height: auto;
  display: flex;
  background: $col1;
  flex-direction: column;
  border-radius: var(--spacing-3);
}

.allregions {
  padding: var(--spacing-7) !important;
}

.middleregion {
  border-top: var(--grey-600) thin solid;
  border-bottom: var(--grey-600) thin solid;

  .myticket {
    border-bottom: var(--grey-600) thin solid;
  }

  svg {
    color: var(--primary-3) !important;
  }
}

.bottombutton {
  background: rgba(255, 255, 255, 0.05) !important;
  min-width: 88px;
  min-height: 88px;
  border-radius: var(--spacing-2);
  &:hover {
    text-decoration: none;
    background: rgba(255, 255, 255, 0) !important;
  }

  svg {
    color: var(--primary-1) !important;
  }
}

.text {
  font-size: 12px !important;
}
