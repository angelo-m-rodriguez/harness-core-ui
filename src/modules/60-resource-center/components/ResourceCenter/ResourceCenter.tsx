import React from 'react'
import { Icon, Text, Layout, Color, IconName } from '@wings-software/uicore'

import cx from 'classnames'
import resourceImage from './images/resource-center.svg'
import css from './ResourceCenter.module.scss'
const getButton = (buttonText: string, buttonIcon: IconName) => {
  return (
    <Layout.Vertical
      flex={{ align: 'center-center' }}
      spacing="small"
      padding={'small'}
      className={cx(css.bottombutton)}
    >
      <Icon name={buttonIcon} size={24} />
      <Text
        font={{ weight: 'semi-bold', align: 'center' }}
        padding={{ bottom: 'xsmall' }}
        color={Color.PRIMARY_3}
        className={css.text}
      >
        {buttonText}
      </Text>
    </Layout.Vertical>
  )
}

export const ResourceCenter = () => {
  return (
    <div className={css.resourceCenter}>
      <Layout.Vertical flex={{ alignItems: 'baseline' }} className={css.allregions}>
        <Text color={Color.WHITE} padding={{ bottom: 'medium' }}>
          RESOURCE CENTER
        </Text>
        <img src={resourceImage} height={106} />
      </Layout.Vertical>
      <Layout.Vertical className={cx(css.allregions, css.middleregion)}>
        <Layout.Horizontal
          padding={{ bottom: 'medium' }}
          flex={{ justifyContent: 'space-between' }}
          className={css.myticket}
        >
          <Layout.Vertical>
            <Text font={{ weight: 'semi-bold', size: 'medium' }} padding={{ bottom: 'xsmall' }} color={Color.PRIMARY_3}>
              {'My Tickets'}
            </Text>
            <Text
              font={{ weight: 'semi-bold' }}
              padding={{ bottom: 'xsmall' }}
              color={Color.WHITE}
              className={css.text}
            >
              {'View and manage your tickets'}
            </Text>
          </Layout.Vertical>
          <Icon name={'chevron-right'} width={6} height={12} />
        </Layout.Horizontal>
        <Layout.Horizontal padding={{ top: 'medium' }} flex={{ justifyContent: 'space-between' }}>
          <Layout.Vertical>
            <Text font={{ weight: 'semi-bold', size: 'medium' }} padding={{ bottom: 'xsmall' }} color={Color.PRIMARY_3}>
              {'Submit a ticket'}
            </Text>
            <Text
              font={{ weight: 'semi-bold' }}
              padding={{ bottom: 'xsmall' }}
              color={Color.WHITE}
              className={css.text}
            >
              {'Get help from our expert system'}
            </Text>
          </Layout.Vertical>
          <Icon name={'chevron-right'} width={6} height={12} />
        </Layout.Horizontal>
      </Layout.Vertical>
      <Layout.Vertical className={css.allregions}>
        <Text font={{ weight: 'semi-bold' }} padding={{ bottom: 'medium' }} color={Color.WHITE} className={css.text}>
          {'Other useful resources'}
        </Text>
        <Layout.Horizontal flex={{ justifyContent: 'space-around' }}>
          {getButton('Search', 'thinner-search')}
          {getButton('Docs', 'file')}
          {getButton('Community', 'people')}
          {getButton('Site Status', 'new-notification')}
        </Layout.Horizontal>
      </Layout.Vertical>
    </div>
  )
}
