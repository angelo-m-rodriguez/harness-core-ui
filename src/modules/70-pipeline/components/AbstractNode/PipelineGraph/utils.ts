import type { RefObject } from 'react'
import type { StageElementWrapperConfig } from 'services/cd-ng'
const INITIAL_ZOOM_LEVEL = 1
const ZOOM_INC_DEC_LEVEL = 0.1
const getFinalSVGArrowPath = (parentId: string, id1 = '', id2 = '', isParallelNode = false): string => {
  if (!parentId) {
    return ''
  }
  const node1 = getComputedPosition(parentId, id1)
  const node2 = getComputedPosition(parentId, id2)
  if (!node1 || !node2) {
    return ''
  }

  const node1VerticalMid = node1.top + node1.height / 2
  const node2VerticalMid = node2.top + node2.height / 2

  const startPoint = `${node1.right},${node1VerticalMid}`
  const horizontalMid = Math.abs((node1.right + node2.left) / 2)
  const endPoint = `${node2.left},${node2VerticalMid}`

  if (node2.y < node1.y) {
    //  child node is at top
    const curveLeftToTop = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid - 20}`
    const curveBottomToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToTop} 
    L${horizontalMid},${node2VerticalMid + 20} ${curveBottomToRight} L${endPoint}`
  } else if (node1.y === node2.y) {
    // both nodes are at same level vertically
    return `M${startPoint}  L${endPoint} `
  } else {
    //  child node is at bottom
    const curveLeftToBottom = `Q${horizontalMid},${node1VerticalMid} ${horizontalMid},${node1VerticalMid + 20}`

    const curveTopToRight = `Q${horizontalMid},${node2VerticalMid} ${horizontalMid + 20},${node2VerticalMid}`

    if (isParallelNode) {
      const updatedStart = node1.left - 45 // new start point
      const parallelLinkStart = `${
        updatedStart // half of link length
      },${node1VerticalMid}`

      const curveLBparallel = `Q${updatedStart + 20},${node1VerticalMid} ${updatedStart + 20},${node1VerticalMid + 20} `
      const curveTRparallel = `Q${updatedStart + 20},${node2VerticalMid} ${updatedStart + 40},${node2VerticalMid}`

      return `M${parallelLinkStart} 
      ${curveLBparallel} 
      L${updatedStart + 20},${node2VerticalMid - 20} 
      ${curveTRparallel} 
      L${node2.left},${node2VerticalMid} 
      M${node2.right},${node2VerticalMid} 
      L${node2.right + 10},${node2VerticalMid}
      Q${node2.right + 25},${node2VerticalMid} ${node2.right + 25},${node2VerticalMid - 20}      
      L${node2.right + 25},${node1VerticalMid + 20}
      Q${node2.right + 25},${node1VerticalMid} ${node2.right + 40},${node1VerticalMid}
      `
    }
    return `M${startPoint} L${horizontalMid - 20},${node1VerticalMid} ${curveLeftToBottom} 
    L${horizontalMid},${node2VerticalMid - 20} ${curveTopToRight} L${endPoint}`
  }
}

const getComputedPosition = (parentId: string, childId: string): DOMRect => {
  const parentPos = document.getElementById(parentId)?.getBoundingClientRect() as DOMRect
  const childPos = document.getElementById(childId)?.getBoundingClientRect() as DOMRect
  const updatedTop = childPos.top - parentPos.top
  const updatedLeft = childPos.left - parentPos.left
  const updatedRight = updatedLeft + childPos.width
  const updatedBottom = updatedTop + childPos.height
  const updatedPos: DOMRect = {
    ...childPos,
    left: updatedLeft,
    top: updatedTop,
    right: updatedRight,
    bottom: updatedBottom,
    width: childPos.width,
    height: childPos.height,
    x: childPos.x,
    y: childPos.y
  }
  return updatedPos
}

const setupDragEventListeners = (canvasRef: RefObject<HTMLDivElement>): (() => void) => {
  let offset = [0, 0]
  const divOverlay = (canvasRef.current as HTMLDivElement).parentElement as HTMLDivElement
  const canvas = canvasRef.current as HTMLDivElement
  let isDown = false

  const onMouseMove = (e: any): void => {
    e.preventDefault()
    if (isDown) {
      canvas.style.left = e.clientX + offset[0] + 'px'
      canvas.style.top = e.clientY + offset[1] + 'px'
    }
  }

  const onMouseDown = (e: any): void => {
    isDown = true
    offset = [canvas.offsetLeft - e.clientX, canvas.offsetTop - e.clientY]
  }
  const onMouseUp = (): void => {
    isDown = false
  }

  divOverlay.addEventListener('mousedown', onMouseDown, true)
  document.addEventListener('mouseup', onMouseUp, true)
  document.addEventListener('mousemove', onMouseMove, true)
  return () => {
    divOverlay.removeEventListener('mousedown', onMouseDown)
    document.removeEventListener('mouseup', onMouseUp)
    document.removeEventListener('mousemove', onMouseMove)
  }
}

const getSVGLinksFromPipeline = (stages?: StageElementWrapperConfig[], resultArr: string[] = []): string[] => {
  let prevElement: StageElementWrapperConfig
  stages?.forEach(stage => {
    if (!prevElement) {
      prevElement = stage
      return
    }
    if (stage.parallel) {
      getParallelNodeLinks(stage.parallel, prevElement, resultArr)
    } else {
      resultArr.push(
        getFinalSVGArrowPath('tree-container', prevElement.stage?.identifier as string, stage.stage?.identifier, false)
      )
    }
    prevElement = stage
  })
  return resultArr
}
const getParallelNodeLinks = (
  stages: StageElementWrapperConfig[],
  prevStage: StageElementWrapperConfig,
  resultArr: string[] = []
): void => {
  const [firstStage, ...restStages] = stages
  resultArr.push(
    getFinalSVGArrowPath('tree-container', prevStage.stage?.identifier as string, firstStage.stage?.identifier, false)
  )
  restStages?.forEach(stage => {
    resultArr.push(
      getFinalSVGArrowPath('tree-container', firstStage.stage?.identifier as string, stage.stage?.identifier, true)
    )
  })
}

const getScaleToFitValue = (elm: HTMLElement, paddingFromBottom = 20): number => {
  return (
    1 /
    Math.max(
      elm.clientWidth / window.innerWidth,
      elm.clientHeight / (window.innerHeight - elm.offsetTop - paddingFromBottom)
    )
  )
}
export {
  getFinalSVGArrowPath,
  setupDragEventListeners,
  getSVGLinksFromPipeline,
  ZOOM_INC_DEC_LEVEL,
  INITIAL_ZOOM_LEVEL,
  getScaleToFitValue
}
