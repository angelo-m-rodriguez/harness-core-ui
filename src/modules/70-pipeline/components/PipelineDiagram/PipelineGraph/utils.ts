import { defaultTo } from 'lodash-es'
import type { PipelineGraphState } from '../types'

const shouldAttachRef = (
  intersectingIndex: number,
  isCurrentChildLast: boolean,
  indexRelativeToParent: number
): boolean => {
  return intersectingIndex === -1 // intersecting index is not set
    ? isCurrentChildLast // current child is not the last child in parallel
    : intersectingIndex === indexRelativeToParent // intersecting index is not equal to indexRelativeToParent (i.e index+1 as parent is 0)
}

const shouldRenderGroupNode = (attachRef: boolean, isCurrentChildLast: boolean): boolean => {
  return attachRef && !isCurrentChildLast // render group node is ref is attached and current node is not the last child
}

const isFirstNodeAGroupNode = (intersectingIndex: number, collapseOnIntersect = false, childLength = 0): boolean =>
  intersectingIndex === 0 && collapseOnIntersect && childLength > 0 // if intersecting index is 0 and collapseOnIntersect is enabled

const showChildNode = (indexRelativeToParent: number, intersectingIndex: number): boolean => {
  return !(indexRelativeToParent > intersectingIndex && intersectingIndex !== -1) // child index is greater than intersectingIndex and intersectingIndex is set to some value
}

const isNodeParallel = (node: PipelineGraphState): boolean => defaultTo(node?.children?.length, 0) > 0

export { shouldAttachRef, shouldRenderGroupNode, isFirstNodeAGroupNode, showChildNode, isNodeParallel }
