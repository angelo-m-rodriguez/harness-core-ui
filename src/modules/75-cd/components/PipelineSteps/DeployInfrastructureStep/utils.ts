/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType, SelectOption } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface PipelineInfrastructureV2 extends PipelineInfrastructure {
  environmentOrEnvGroupRef?: SelectOption
  environmentGroup?: any
  environmentRef2?: any
  infrastructureRef?: any
}

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructureV2
  onUpdate?: (data: PipelineInfrastructureV2) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructureV2
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructureV2>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructureV2): boolean {
  if (
    getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.environmentOrEnvGroupRef)
  ) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
