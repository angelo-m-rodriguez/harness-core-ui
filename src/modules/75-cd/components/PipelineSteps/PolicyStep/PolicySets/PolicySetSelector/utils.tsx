/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { FontVariation, Layout, Text } from '@harness/uicore'
import { useStrings } from 'framework/strings'

export const PolicySetSelectorDialogTitle = (): JSX.Element => {
  const { getString } = useStrings()
  return (
    <Layout.Horizontal
      spacing="xsmall"
      padding={{ top: 'xxlarge', left: 'medium', right: 'large', bottom: 'medium' }}
      flex={{ justifyContent: 'space-between' }}
    >
      <Layout.Vertical>
        <Text font={{ variation: FontVariation.H3 }}>{getString('common.policiesSets.selectPolicySet')}</Text>
      </Layout.Vertical>
    </Layout.Horizontal>
  )
}
