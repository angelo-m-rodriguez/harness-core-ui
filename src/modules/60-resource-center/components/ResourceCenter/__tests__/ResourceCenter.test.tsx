import { render } from '@testing-library/react'
import React from 'react'
import { TestWrapper } from '@common/utils/testUtils'
import { ResourceCenter } from '@resource-center/components/ResourceCenter/ResourceCenter'
describe('ResourceCenter', () => {
  test('Should render resource center properly', () => {
    const { container } = render(
      <TestWrapper>
        <ResourceCenter />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
