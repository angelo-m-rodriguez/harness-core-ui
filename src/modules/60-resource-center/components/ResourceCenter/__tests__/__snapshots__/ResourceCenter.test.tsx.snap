// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCenter Should render resource center properly 1`] = `
<div>
  <div
    class="resourceCenter"
  >
    <div
      class="Layout--vertical StyledProps--main allregions StyledProps--flex StyledProps--flex-alignItems-baseline"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-white StyledProps--padding-bottom-medium"
      >
        resourceCenter.title
      </p>
      <img
        height="106"
        src="test-file-stub"
      />
    </div>
    <div
      class="Layout--vertical StyledProps--main allregions middleregion"
    >
      <div
        class="Layout--horizontal StyledProps--main myticket StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-justifyContent-space-between"
      >
        <div
          class="Layout--vertical StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--font-size-medium StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-primary3"
          >
            resourceCenter.ticketmenu.tickets
          </p>
          <p
            class="StyledProps--font StyledProps--main text StyledProps--font-variation-body2 StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-white"
          >
            resourceCenter.ticketmenu.ticketsDesc
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-chevron-right StyledProps--main"
          icon="chevron-right"
          style="width: 6px; height: 12px;"
        >
          <svg
            data-icon="chevron-right"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-right
            </desc>
            <path
              d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="Layout--horizontal StyledProps--main StyledProps--padding-top-medium StyledProps--flex StyledProps--flex-justifyContent-space-between"
      >
        <div
          class="Layout--vertical StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--font-size-medium StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-primary3"
          >
            resourceCenter.ticketmenu.submit
          </p>
          <p
            class="StyledProps--font StyledProps--main text StyledProps--font-variation-body2 StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-white"
          >
            resourceCenter.ticketmenu.submitDesc
          </p>
        </div>
        <span
          class="bp3-icon bp3-icon-chevron-right StyledProps--main"
          icon="chevron-right"
          style="width: 6px; height: 12px;"
        >
          <svg
            data-icon="chevron-right"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-right
            </desc>
            <path
              d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="Layout--vertical StyledProps--main allregions"
    >
      <p
        class="StyledProps--font StyledProps--main text StyledProps--font-weight-semi-bold StyledProps--padding-bottom-medium StyledProps--color StyledProps--color-white"
      >
        resourceCenter.bottomlayout.desc
      </p>
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-around"
      >
        <a
          href="resourceCenter.bottomlayout.search.link"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-small StyledProps--main bottombutton StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-small"
          >
            <span
              class="bp3-icon bp3-icon-resourceCenter.bottomlayout.search.icon StyledProps--main"
              icon="resourceCenter.bottomlayout.search.icon"
            >
              <svg
                data-icon="resourceCenter.bottomlayout.search.icon"
                height="24"
                viewBox="0 0 20 20"
                width="24"
              >
                <desc>
                  resourceCenter.bottomlayout.search.icon
                </desc>
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main text StyledProps--font-variation-body2 StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-primary3"
            >
              search
            </p>
          </div>
        </a>
        <a
          href="resourceCenter.bottomlayout.docs.link"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-small StyledProps--main bottombutton StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-small"
          >
            <span
              class="bp3-icon bp3-icon-resourceCenter.bottomlayout.docs.icon StyledProps--main"
              icon="resourceCenter.bottomlayout.docs.icon"
            >
              <svg
                data-icon="resourceCenter.bottomlayout.docs.icon"
                height="24"
                viewBox="0 0 20 20"
                width="24"
              >
                <desc>
                  resourceCenter.bottomlayout.docs.icon
                </desc>
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main text StyledProps--font-variation-body2 StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-primary3"
            >
              resourceCenter.bottomlayout.docs.text
            </p>
          </div>
        </a>
        <a
          href="resourceCenter.bottomlayout.community.link"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-small StyledProps--main bottombutton StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-small"
          >
            <span
              class="bp3-icon bp3-icon-resourceCenter.bottomlayout.community.icon StyledProps--main"
              icon="resourceCenter.bottomlayout.community.icon"
            >
              <svg
                data-icon="resourceCenter.bottomlayout.community.icon"
                height="24"
                viewBox="0 0 20 20"
                width="24"
              >
                <desc>
                  resourceCenter.bottomlayout.community.icon
                </desc>
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main text StyledProps--font-variation-body2 StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-primary3"
            >
              authSettings.cdCommunityPlan.communityTitle
            </p>
          </div>
        </a>
        <a
          href="resourceCenter.bottomlayout.sitestatus.link"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="Layout--vertical Layout--layout-spacing-small StyledProps--main bottombutton StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-small"
          >
            <span
              class="bp3-icon bp3-icon-resourceCenter.bottomlayout.sitestatus.icon StyledProps--main"
              icon="resourceCenter.bottomlayout.sitestatus.icon"
            >
              <svg
                data-icon="resourceCenter.bottomlayout.sitestatus.icon"
                height="24"
                viewBox="0 0 20 20"
                width="24"
              >
                <desc>
                  resourceCenter.bottomlayout.sitestatus.icon
                </desc>
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main text StyledProps--font-variation-body2 StyledProps--padding-bottom-xsmall StyledProps--color StyledProps--color-primary3"
            >
              resourceCenter.bottomlayout.sitestatus.text
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
